docs/hugo/content/architecture-decision-records/ADR-2022-01-Reconciler-Extensions.md

```diff
@@ -0,0 +1,106 @@
+# Reconciler Extensions
+
+## Context
+
+We are discovering inconsistencies in the way different Azure Resource Providers behave.
+
+These inconsistencies are anticipated. Given the number of disparate teams independently implementing their providers and given the way the ARM guidance has changed over time, differences in behaviour are to be expected.
+
+For example, the Azure Redis Resource Provider can return a HTTP conflict (409) with a retry message indicating that it's not a fatal error but a transient one.
+
+Currently we handle this with special case behaviour in `error_classifier.go` (see [PR#2008](https://github.com/Azure/azure-service-operator/pull/2008) for details).
+
+However, directly changing our generic reconciler is a problem. The growing complexity makes this unsustainable, and there is a risk that changes made to support a new resource type will break an existing one.
```

Comment:
minor: complexity and number of resources

Reply:
Reworded.